code/game/gamemodes/objective.dm

```diff
@@ -164,6 +164,35 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return FALSE
 	return TRUE
 
+/datum/objective/assassinateonce
+	name = "Assassinate once"
+	martyr_compatible = TRUE
+	var/won = FALSE
+
+/datum/objective/assassinateonce/find_target(list/target_blacklist)
+	..()
+	if(target?.current)
+		explanation_text = "Teach [target.current.real_name], the [target.assigned_role], a lesson they will not forget. The target only needs to die once for success."
+		RegisterSignal(target, COMSIG_MOB_DEATH, PROC_REF(check_midround_completion))
```

Comment:
You should probably change all of these death signals to be death signals and deleting signals
```suggestion
		RegisterSignal(target, list(COMSIG_MOB_DEATH, COMSIG_PARENT_QDELETING), PROC_REF(check_midround_completion))
```